src/desktop_notifier/util/logger.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+"""
+Centralizes logging for desktop-notifier. Configure your own logging with:
+
+.. code-block:: python
+    import logging
+    from desktop_notifier.util.logger import Logger
+    Logger.set_logger("my_logger")
+
+Then:
+
+.. code-block:: python
+    from desktop_notifier.util.logger import Logger
+    Logger.logger().info("This will be logged using the logger named 'my_logger'")
+"""
+import logging
+
+from desktop_notifier.util.pkg_metadata_helper import __DESKTOP_NOTIFIER_PACKAGE_NAME__
+
+DEFAULT_LOGGER = logging.getLogger(__DESKTOP_NOTIFIER_PACKAGE_NAME__)
+
+
+class Logger:
+    _logger: logging.Logger | None = None
+
+    @classmethod
+    def logger(cls) -> logging.Logger:
```

Comment:
i defined `logger()` as a class method so that `_logger` could be painlessly swapped out underneath it. also decided to just avoid any kind of singleton shenanigans or any kind of overloading `Logger.info()`, `Logger.warning()`, etc. but there are many ways this could be done and tbh i think it's really just a stylistic choice.